src/main/java/ubc/pavlab/rdp/repositories/UserRepository.java

```diff
@@ -38,8 +38,8 @@ public interface UserRepository extends JpaRepository<User, Integer> {
     User findByEmailIgnoreCase( @Param("email") String email );
 
     @QueryHints(@QueryHint(name = "org.hibernate.cacheable", value = "true"))
-    @Query("select u from User u where u.profile.lastName is not NULL and u.profile.lastName <> ''")
-    Collection<User> findAllWithNonEmptyProfileLastName();
+    @Query("select u from User u where u.profile.lastName is not NULL and u.profile.lastName <> '' and u.profile.privacyLevel <> ubc.pavlab.rdp.model.enums.PrivacyLevelType.PRIVATE")
+    Collection<User> findAllWithNonEmptyProfileLastNameAndProfilePrivacyLevelNotPrivate();
```

Comment:
This could use a strictly greater or equal than with a binding parameter to choose between shared or public depending on the logged in status.